externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/kyuubi/SQLOperationListener.scala

```diff
@@ -134,9 +140,15 @@ class SQLOperationListener(
 
   override def onStageCompleted(stageCompleted: SparkListenerStageCompleted): Unit = {
     val stageInfo = stageCompleted.stageInfo
+    val stageId = stageInfo.stageId
     val stageAttempt = SparkStageAttempt(stageInfo.stageId, stageInfo.attemptNumber())
     activeStages.synchronized {
       if (activeStages.remove(stageAttempt) != null) {
+        activeJobs.asScala.foreach(item => {
+          if (item._2.stageIds.contains(stageId)) {
+            item._2.numCompleteStages.getAndIncrement()
```

Comment:
Add one when the stage is successful.

https://github.com/apache/spark/blob/28961a6ce001e0c25c780a39a726fdd825542cee/core/src/main/scala/org/apache/spark/status/AppStatusListener.scala#L827-L836